providers/storetheindex.go

```diff
@@ -0,0 +1,52 @@
+package providers
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/libp2p/go-libp2p-core/peer"
+	"github.com/libp2p/go-libp2p-kad-dht/providers"
+	"github.com/libp2p/hydra-booster/metrics"
+	"github.com/libp2p/hydra-booster/providers/storetheindex"
+	"github.com/multiformats/go-multihash"
+	"go.opencensus.io/stats"
+	"go.opencensus.io/tag"
+)
+
+func StoreTheIndexProvider(endpointURL string, timeout time.Duration) (providers.ProviderStore, error) {
+	c, err := storetheindex.New(endpointURL, storetheindex.WithHTTPClient(&http.Client{Timeout: timeout}))
+	if err != nil {
+		return nil, err
+	}
+	return &StoreTheIndexProviderStore{c: c}, nil
+}
+
+type StoreTheIndexProviderStore struct {
+	c storetheindex.Client
+}
+
+func (s *StoreTheIndexProviderStore) AddProvider(ctx context.Context, key []byte, prov peer.AddrInfo) error {
+	return fmt.Errorf("adding providers not supported")
+}
+
+func (s *StoreTheIndexProviderStore) GetProviders(ctx context.Context, key []byte) ([]peer.AddrInfo, error) {
+	h, err := multihash.Cast(key)
+	if err != nil {
+		return nil, err
+	}
+	t0 := time.Now()
+	infos, err := s.c.FindProviders(ctx, h)
+	dur := time.Now().Sub(t0)
+	recordSTIFindProvsComplete(ctx, statusFromErr(err), metrics.STIFindProvsDuration.M(float64(dur)))
```

Comment:
Can we add HTTP status code as a metric tag so we can understand the failures with more granularity (e.g. 4xx vs 5xx) ?